spec/pharos/addons/host_upgrades_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'pharos/addon'
+require "./addons/host-upgrades/addon"
+
+describe Pharos::Addons::HostUpgrades do
+  let(:cluster_config) { Pharos::Config.load(
+    hosts: [ {role: 'master', address: '192.0.2.1'} ],
+  ) }
+  let(:config) { { } }
+  let(:cpu_arch) { double(:cpu_arch ) }
+  let(:master) { double(:host, address: '192.0.2.1') }
+
+  subject do
+    described_class.new({enabled: true}.merge(config), enabled: true, master: master, cpu_arch: cpu_arch, cluster_config: cluster_config)
+  end
+
+  describe "#validate" do
+    it "fails with missing schedule" do
+      result = described_class.validate({enabled: true})
+
+      expect(result).to_not be_success
+      expect(result.errors.dig(:schedule)).to eq ["is missing"]
+    end
+
+    describe 'schedule' do
+      it "fails with empty schedule" do
+        result = described_class.validate({enabled: true, schedule: ''})
+
+        expect(result).to_not be_success
+        expect(result.errors.dig(:schedule)).to eq ["must be filled"]
+      end
+
+      it "fails with invalid schedule" do
+        result = described_class.validate({enabled: true, schedule: '3'})
+
+        expect(result).to_not be_success
+        expect(result.errors.dig(:schedule)).to match [/is not a valid crontab/]
+      end
+
+      it "fails with short cron schedule" do
+        result = described_class.validate({enabled: true, schedule: '0 * * *'})
+
+        expect(result).to_not be_success
+        expect(result.errors.dig(:schedule)).to match [/is not a valid crontab/]
+      end
+
+      it "fails with long cron schedule" do
+        result = described_class.validate({enabled: true, schedule: '0 0 0 * * *'})
+
+        expect(result).to_not be_success
```

Comment:
Having a hard time figuring out how to get `Fugit::Cron.parse` to reject or normalize the extended seconds syntax without monkey-patching it...

```
  1) Pharos::Addons::HostUpgrades#validate schedule fails with long cron schedule
     Failure/Error: expect(result).to_not be_success
       expected `#<Dry::Validation::Result output={:enabled=>true, :schedule=>"0 0 0 * * *"} errors={}>.success?` to return false, got true
     # ./spec/pharos/addons/host_upgrades_spec.rb:49:in `block (4 levels) in <top (required)>'
     # ./spec/spec_helper.rb:121:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:111:in `block (2 levels) in <top (required)>'

```

Reply:
Hacked this in with a trivial monkeypatch to add a missing `attr_reader`... validating/normalizing the cron schedule was more difficult than expected this way.